content/ADR-164-quests.poc.md

```diff
@@ -0,0 +1,211 @@
+
+Having a Quests System can solve part of the problem, giving users a reason to stay and have a nice time. This feature would also allow content creators to be creative and increase engagement.
+
+The last sentence is important to remark on since it is one of the differences to any other Quest System you may know from video games, where the quests are defined by the owners or internal teams. In this case, there should be a focus on creating easy tools for quest designers.
+
+On the other side, the system (backend) should be in charge of tracking and validating the progress and accepting new quest deployments.
+
+Having said that, there are three well-known actors with corresponding components:
+ - Users: Explorer and Kernel 
+ - Designers: Visual tool and/or CLI
+ - The System: Backend
+
+For the sake of fast iteration and discovery, this document will explore the boundaries and scope of a Proof of Concept for the described Quests System.
+
+## Solution Space Exploration
+
+The following sections will describe what would be necessary and alternatives for this Proof of Concept, where designers can submit quests, users experience them and the system expose some metrics. 
+
+### Quest definition
+
+A Quest definition will include steps/tasks required for completion. It would be defined technically speaking as a graph, since it may have a non-linear complexity.
+
+Defining a valid graph, as it may require some conditions:
+- it should have only one start node.
+- it should have at least one end node.  
+- it may have cycles but not endless loops.
+- it may have conditional nodes.
+
+### Explorer
+
+Minimal UI for quest tracking and receive real-time notifications about the progress:
+
+- Users should have a UI where to see the current quests (in progress) and the status. 
+- Users should be able to cancel/drop a quest.
+- Users should be able to discover quests in World. (NPC Givers)
+
+This UI may be developed using SDK and defining reusable UI components so every Explorer implementation should not implement custom UIs.
+
+### Kernel
+
+Communication with the system, notifying events that happen in world, and receiving updates, so the Explorer is up to date.
+
+- Kernel should be responsible of receiving and notifiyng updates to Explorer.
+- Kernel should be able to receive requests from Explorer to start a quest and ask for it to the System.
+- Kernel should be able to notify in world events to the System, in order to make progress.
+
+### Actions
+
+A set of predefined actions may be provided to the designers and users, in order to define the Quests steps and make progress. 
+
+- Location
+- NPC Interaction
+- Play Emote
+- TBD
+
+### Quest Designer
+
+Build a tool/CLI to easily design Quests, without conflicting with existing definitions.
```

Comment:
Add something about personas here, mainly to introduce the complexity of a person that's not an experienced programmer.